src/nupic/algorithms/Connections.cpp

```diff
@@ -557,6 +557,44 @@ void Connections::bumpSegment(const Segment segment, const Permanence delta) {
 }
 
 
+void Connections::destroyMinPermanenceSynapses(Segment segment, Int nDestroy,
+                                         const vector<CellIdx> &excludeCells) {
+  // Don't destroy any cells that are in excludeCells.
```

Comment:
do `if(nDestroy ==0) return; ` //or <= if not unsigned. 
to skip the binary_search below